File: Euclid/Messages/Outgoing/ALLUNITS.cs
using Euclid.Game;
using Euclid.Network.Streams.Util;
using System.Collections.Generic;

namespace Euclid.Messages.Outgoing
{
    internal class ALLUNITS : IMessageComposer
    {
        private List<Room> rooms;

        public ALLUNITS(List<Room> rooms)
        {
            this.rooms = rooms;
        }

        public override void Write()
        {
            foreach (var room in rooms)
            {
                Data.Add(new ArgumentEntry(room.Data.Name));
                Data.Add(new DelimeterEntry(room.Data.UsersNow, ","));
                Data.Add(new DelimeterEntry(room.Data.UsersMax, ","));
                Data.Add(new DelimeterEntry(room.Address.IpAddress, ","));
                Data.Add(new DelimeterEntry(room.Address.IpAddress, "/"));
                Data.Add(new DelimeterEntry(room.Address.Port, ","));
                Data.Add(new DelimeterEntry(room.Data.Name, ","));
                Data.Add(new DelimeterEntry(room.Data.Description, ","));
                Data.Add(new DelimeterEntry(room.Data.UsersNow, ","));
                Data.Add(new DelimeterEntry(room.Data.UsersMax, ","));
                Data.Add(new DelimeterEntry(room.Model.Data.Model, ","));
            }

            //this.m_Data.Add($"Main Lobby,0,40,127.0.0.1/127.0.0.1,{GameServer.Instance.MainPort + 1},Main Lobby" + (char)9 + "lobby,0,40,lobby_a" + (char)13);
            //this.m_Data.Add($"Median Lobby,0,40,127.0.0.1/127.0.0.1,{GameServer.Instance.MainPort + 1},Median Lobby" + (char)9 + "lobby,0,40,floorlobby_b" + (char)13);
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/ALLUNITS.java
package euclid.messages.outgoing;

import euclid.game.room.Room;
import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;
import euclid.network.streams.util.DelimeterEntry;

import java.util.List;

public class ALLUNITS extends IMessageComposer {

    private final List<Room> rooms;

    public ALLUNITS(List<Room> rooms) {
        this.rooms = rooms;
    }

    @Override
    public void write() {
        for (Room room : rooms) {
            getData().add(new ArgumentEntry(room.getData().getName()));
            getData().add(new DelimeterEntry(room.getData().getUsersNow(), ","));
            getData().add(new DelimeterEntry(room.getData().getUsersMax(), ","));
            getData().add(new DelimeterEntry(room.getAddress().getIpAddress(), ","));
            getData().add(new DelimeterEntry(room.getAddress().getIpAddress(), "/"));
            getData().add(new DelimeterEntry(room.getAddress().getPort(), ","));
            getData().add(new DelimeterEntry(room.getData().getName(), ","));
            getData().add(new DelimeterEntry(room.getData().getDescription(), ","));
            getData().add(new DelimeterEntry(room.getData().getUsersNow(), ","));
            getData().add(new DelimeterEntry(room.getData().getUsersMax(), ","));
            getData().add(new DelimeterEntry(room.getModel().getData().getModel(), ","));
        }
    }
}
